From a3468ed19d6e3f30dff97aec9e0cbdff781fad86 Mon Sep 17 00:00:00 2001
From: Ramax Lo <ramax.lo@mediatek.com>
Date: Thu, 20 Feb 2025 10:41:51 +0800
Subject: [PATCH] GENIO: drivers: Strip path components

The host tool conmakehash outputs absolute paths in the generated
files, which causes warnings of the package QA process of Yocto.
This commit modifies the source to only output basenames of paths.

Change-Id: I2960019174103bd3ced7cc80a2833e140ed5d151
Signed-off-by: Ramax Lo <ramax.lo@mediatek.com>
---
 drivers/tty/vt/conmakehash.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/tty/vt/conmakehash.c b/drivers/tty/vt/conmakehash.c
index cddd789fe46e..89474be2a769 100644
--- a/drivers/tty/vt/conmakehash.c
+++ b/drivers/tty/vt/conmakehash.c
@@ -16,6 +16,7 @@
 #include <sysexits.h>
 #include <string.h>
 #include <ctype.h>
+#include <libgen.h>
 
 #define MAX_FONTLEN 256
 
@@ -82,6 +83,7 @@ int main(int argc, char *argv[])
   int i, nuni, nent;
   int fp0, fp1, un0, un1;
   char *p, *p1;
+  char tmpbuf[65536];
 
   if ( argc < 2 || argc > 5 )
     usage(argv[0]);
@@ -242,6 +244,8 @@ int main(int argc, char *argv[])
   for ( i = 0 ; i < fontlen ; i++ )
     nuni += unicount[i];
 
+  strncpy(tmpbuf, argv[1], sizeof(tmpbuf) - 1);
+  tmpbuf[sizeof(tmpbuf) - 1] = '\0';
   printf("\
 /*\n\
  * Do not edit this file; it was automatically generated by\n\
@@ -253,7 +257,7 @@ int main(int argc, char *argv[])
 #include <linux/types.h>\n\
 \n\
 u8 dfont_unicount[%d] = \n\
-{\n\t", argv[1], fontlen);
+{\n\t", basename(tmpbuf), fontlen);
 
   for ( i = 0 ; i < fontlen ; i++ )
     {
-- 
2.45.2

